bindata/network/multus/multus.yaml

```diff
@@ -66,24 +71,38 @@ data:
         fi
       ;;
       *)
-        echo "ERROR: RHEL Major Version Unsupported, rhelmajor=${rhelmajor}"
+        log "ERROR: RHEL Major Version Unsupported, rhelmajor=${rhelmajor}"
       ;;
     esac
 
     # When it doesn't exist, fall back to the original directory.
     if [ "$founddir" == false ]; then
-      echo "Source directory unavailable for OS version: ${rhelmajor}"
+      log "Source directory unavailable for OS version: ${rhelmajor}"
       sourcedir=$DEFAULT_SOURCE_DIRECTORY
     fi
 
-    cp -rf ${sourcedir}* $DESTINATION_DIRECTORY
-
+    # Use a subdirectory called "upgrade" so we can atomically move fully copied files.
+    UPGRADE_DIRECTORY=${DESTINATION_DIRECTORY}upgrade_$(uuidgen)
+    rm -Rf $UPGRADE_DIRECTORY
+    mkdir -p $UPGRADE_DIRECTORY
+    cp -rf ${sourcedir}* $UPGRADE_DIRECTORY
     if [ $? -eq 0 ]; then
-      echo "Successfully copied files in ${sourcedir} to $DESTINATION_DIRECTORY"
+      log "Successfully copied files in ${sourcedir} to $UPGRADE_DIRECTORY"
     else
-      echo "Failed to copy files in ${sourcedir} to $DESTINATION_DIRECTORY"
+      log "Failed to copy files in ${sourcedir} to $UPGRADE_DIRECTORY"
+      rm -Rf $UPGRADE_DIRECTORY
       exit 1
     fi
+    mv $UPGRADE_DIRECTORY/* ${DESTINATION_DIRECTORY}/
```

Comment:
does `mv` require a `-f` like `cp` did?

Reply:
yeah, this makes sense. I believe mv will prompt if the destination doesn't have a write flag, or if somehow it gets aliased to `mv -i` in the future, this would prevent it. Good call